### PR TITLE
fix test VerifyDataAvailableAtIoTHub

### PR DESCRIPTION
**Pull request by dacolgit**
Bug: [E2E orchestrated tests for OPC Publisher are failing](https://msazure.visualstudio.com/One/_workitems/edit/12373663)

### CODE REVIEW

**Review by cristipogacean — DISMISSED** [review]
:shipit:

File: e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs [hunk 1]
```diff
@@ -134,7 +134,7 @@ public class A_PublishSingleNodeOrchestratedTestTheory
             var route = string.Format(TestConstants.APIRoutes.PublisherStartFormat, _context.OpcUaEndpointId);
             var body = new {
                 item = new {
-                    nodeId = simulatedOpcServer.Values.First().OpcNodes.First().Id,
+                    nodeId = simulatedOpcServer.Values.First().OpcNodes.Skip(2).First().Id,
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs]: @dacolgit, can you please also filter out nodes with `bad` in the name before selecting them ? We had similar problems in other tests where bad nodes were selected for monitoring. That causes failures as those bad nodes do drop values by design. If you check our E2E test codes you will find a line like this in other places as well:

```
.Where(node => !node.Id.Contains("bad", StringComparison.OrdinalIgnoreCase))
```

Comment (dacolgit) [reply e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs]: added

File: e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs [hunk 2]
```diff
@@ -173,7 +175,7 @@ public class A_PublishSingleNodeOrchestratedTestTheory
             Assert.Equal(1, (int)dataSet.dataSetSource.publishedVariables.publishedData.Count);
             Assert.NotEmpty((string)dataSet.dataSetSource.publishedVariables.publishedData[0].publishedVariableNodeId);
             var publishedNodeId = (string)dataSet.dataSetSource.publishedVariables.publishedData[0].publishedVariableNodeId;
-            Assert.Equal(simulatedOpcServer.Values.First().OpcNodes.First().Id, publishedNodeId);
+            Assert.Equal(simulatedOpcServer.Values.First().OpcNodes.Skip(2).First().Id, publishedNodeId);
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs]: @dacolgit, shouldn't we have the same `Where()` statement here as well ?

Comment (dacolgit) [reply e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/A_PublishSingleNodeOrchestratedTestTheory.cs]: yes

**Review by karok2m — APPROVED** [review]

**Review by cristipogacean — APPROVED** [review]

**Pull request merged**